app/src/main/java/com/sun/imageeditor/screen/home/adapter/CollectionCoverPhotoAdapter.kt

```diff
@@ -0,0 +1,81 @@
+package com.sun.imageeditor.screen.home.adapter
+
+import android.graphics.Bitmap
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.imageeditor.R
+import com.sun.imageeditor.data.model.PhotoCollection
+import com.sun.imageeditor.databinding.ItemContainerCollecitonCoverPhotoBinding
+import com.sun.imageeditor.utils.OnItemRecyclerViewClickListener
+import java.util.concurrent.ConcurrentHashMap
+
+class CollectionCoverPhotoAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    private val coverPhotos = mutableListOf<PhotoCollection>()
+    private val cacheCoverPhotoMap = ConcurrentHashMap<String, Bitmap>()
+    private var isLoading = true
+    private var onItemClick: OnItemRecyclerViewClickListener<String>? = null
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            TYPE_COVER_PHOTO -> CoverPhotoViewHolder(ItemContainerCollecitonCoverPhotoBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            ))
+            else -> throw IllegalArgumentException("Invalid view type")
+        }
+    }
+
+    override fun getItemCount(): Int = coverPhotos.size
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is CoverPhotoViewHolder -> holder.bind(coverPhotos[position])
+            // TODO: implement other view holder types
+        }
+    }
+
+    override fun onViewRecycled(holder: RecyclerView.ViewHolder) {
+        super.onViewRecycled(holder)
+        if (isLoading && holder is CoverPhotoViewHolder) holder.onLoading()
+    }
+
+    fun setCoverPhotos(list: MutableList<PhotoCollection>) {
+        coverPhotos.clear()
+        coverPhotos.addAll(list)
+        notifyDataSetChanged()
+    }
+
+    fun addCoverPhotos(list: MutableList<PhotoCollection>) {
```

Comment:
cho anh hỏi, sao đoạn này em không dùng hàm addAll để add toàn bộ list, sau đó gọi notifyItemRangeInserted, mà lại add từng item một rồi notify từng phát một